src/pages/Scheduling/components/ScheduleChargeItemDefinitionSelector.tsx

```diff
@@ -85,77 +119,99 @@ export default function ScheduleChargeItemDefinitionSelector({
-                </div>
-              </div>
-
-              <div>
-                <Label>{t("re_visit_allowed_days")}</Label>
-                <div className="mt-2">
-                  <Input
-                    type="number"
-                    min={0}
-                    value={reVisitDays}
-                    onChange={(e) =>
-                      setReVisitDays(parseInt(e.target.value) || 0)
-                    }
-                    placeholder={t("enter_re_visit_allowed_days")}
-                  />
-                </div>
-              </div>
-
-              <div>
-                <Label>{t("re_visit_consultation_charge")}</Label>
-                <div className="mt-2 flex gap-2 flex-row">
-                  <ChargeItemDefinitionPicker
-                    facilityId={facilityId}
-                    resourceSubType={
-                      ResourceCategorySubType.charge_item_definition_schedule_practitioner
-                    }
-                    value={reVisitCSlug}
-                    onValueChange={setReVisitCSlug}
-                    placeholder={t("select_charge_item_definition")}
-                    className="flex-1"
-                    showCreateButton={true}
-                  />
-                </div>
-              </div>
+        <Form {...form}>
+          <form
+            onSubmit={form.handleSubmit(onSubmit)}
+            className="mt-6 flex flex-col gap-6"
+          >
+            <FormField
+              control={form.control}
+              name="charge_item_definition_slug"
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel>{t("consultation charge")}</FormLabel>
+                  <FormControl>
+                    <ChargeItemDefinitionPicker
+                      facilityId={facilityId}
+                      resourceSubType={
+                        ResourceCategorySubType.charge_item_definition_schedule_practitioner
+                      }
+                      value={field.value}
+                      onValueChange={(value) => field.onChange(value || "")}
+                      placeholder={t("select_charge_item_definition")}
+                      className="flex-1"
+                      showCreateButton={true}
+                    />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+            <FormField
+              control={form.control}
+              name="re_visit_allowed_days"
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel>{t("re_visit_allowed_days")}</FormLabel>
+                  <FormControl>
+                    <Input
+                      type="number"
+                      min={0}
+                      value={field.value}
+                      onChange={(e) => {
+                        const value =
+                          e.target.value === "" ? 0 : parseInt(e.target.value);
```

Comment:
This needs tweaking; check the behavior when you have 0 and try to change it. It won't allow resetting to empty value and then typing in something new.

Reply:
We can use onBlur instead to force 0 in case user decides to leave input empty 🤔